nss/aad_auth/mock_test.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import "fmt"
+
+const defaultEntry string = "myuser%d@domain.com:x:%d:%d::/home/myuser%d@domain.com:/bin/bash"
+
+type entryMock struct {
+	str string
+}
+
+func (e entryMock) String() string {
+	return e.str
+}
+
+func getMockEntries(n int) []fmt.Stringer {
```

Comment:
This is nice :) I would just rename it to newMockEntries as you instantiate them :)